src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,194 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.lib.exceptions.DataProcessingException;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE deleted = FALSE AND login = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                return Optional.of(getUser(resultSet));
```

Comment:
you're opening connection inside of connection (getRoles() method) 
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_23.html